App/Conditions/Base.php

```diff
@@ -0,0 +1,514 @@
+<?php
+/**
+ * @author      Wployalty (Alagesan)
+ * @license     http://www.gnu.org/licenses/gpl-2.0.html
+ * @link        https://www.wployalty.net
+ * */
+
+namespace Wljm\App\Conditions;
+defined( 'ABSPATH' ) or die();
+
+use DateTime;
+use Exception;
+use stdClass;
+use Wljm\App\Helpers\Input;
+use Wljm\App\Helpers\Woocommerce;
+abstract class Base {
+	public static $woocommerce_helper, $filter;
+	public $name = null, $rule = null, $label = null, $group = null, $input, $extra_params = array( 'render_saved_condition' => false );
+
+	function __construct() {
```

Comment:
Don't do static variable object in constructor. WPLoyalty we made mistake, we have to fix in wployalty also in later.

$woocommerce_helper and $this->input. Need to change that

Reply:
- Removed $woocommerce_helper and used the getInstance() of Woocommerce in all the required places.
- Removed the $this->input, and not added anywhere, as it nowhere used.